src/Extensions/ServiceFabricGuest.Abstractions/factory/ServiceFabricClientFactory.cs

```diff
@@ -0,0 +1,23 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.Omex.Extensions.ServiceFabricGuest.Abstractions
+{
+	/// <summary>
+	/// Factory for service fabric client wrapper
+	/// </summary>
+	public class ServiceFabricClientFactory
+	{
+		/// <summary>
+		/// Creates service fabric client wrapper instance and returns it
+		/// </summary>
+		/// <param name="options">REST client options</param>
+		/// <returns></returns>
+		public static IServiceFabricClientWrapper CreateServiceFabricClientWrapper(IOptions<ServiceFabricRestClientOptions> options)
```

Comment:
This method would be used when there is no way of getting client wrapper from DI, it's better to use `ServiceFabricRestClientOptions` directly as a parameter or even ClusterEndpoint directly

Reply:
By passing ServiceFabricRestClientOptions I wanted hide the way how it is already created (Maybe in the future there will be something other needed that URI of cluster endpoint). I replaced that with ServiceFabricRestClientOptions